Integrations/integration-Exchange2016_Compliance.yml

```diff
@@ -0,0 +1,476 @@
+    }else{
+        $url = "https://" + $server + "/PowerShell"
+        $session = New-PSSession -ConfigurationName Microsoft.Exchange -ConnectionUri $url -Credential $UserCredential -Authentication Basic -AllowRedirection
+    }
+    if (!$session)
+    {
+        "Failed to create remote PS session"
+        return
+    }
+    Import-PSSession $session -CommandName *Compliance* -AllowClobber -DisableNameChecking -Verbose:$false | Out-Null
+    $actionName = $searchName + "_Purge"
+    $actionStatus = Get-ComplianceSearchAction $actionName
+    ""
+    $actionStatus.Status
+    Remove-PSSession $session
+    '''
+
+    TESTCON = '''
+    [CmdletBinding()]
+    Param(
+    [Parameter(Mandatory=$True)]
+    [string]$username,
+    [Parameter(Mandatory=$True)]
+    [string]$password,
+    [Parameter(Mandatory=$True)]
+    [string]$server,
+    [Parameter(Mandatory=$True)]
+    [bool]$unsecure
+    )
+    $errorActionPreference = 'Stop'
+    $WarningPreference = "silentlyContinue"
+    $secpasswd = ConvertTo-SecureString $password -AsPlainText -Force
+    $UserCredential = New-Object System.Management.Automation.PSCredential ($username, $secpasswd)
+    try{
+        if($unsecure){
+            $url = "http://" + $server + "/PowerShell"
+            $session = New-PSSession -ConfigurationName Microsoft.Exchange -ConnectionUri $url -Credential $UserCredential -Authentication Kerberos
+        }else{
+            $url = "https://" + $server + "/PowerShell"
+            $session = New-PSSession -ConfigurationName Microsoft.Exchange -ConnectionUri $url -Credential $UserCredential -Authentication Basic -AllowRedirection
+        }
+        echo "successful connection"
+    } catch {
+        $e = $_.Exception
+        echo $e.Message
+    }
+    '''
+
+    def prepare_args(d):
+        return dict((k.replace("-", "_"), v) for k, v in d.items())
+
+    def str_to_unicode(obj):
+        if isinstance(obj, dict):
+            obj = {k: str_to_unicode(v) for k,v in obj.iteritems()}
+        elif isinstance(obj, list):
+            obj = map(str_to_unicode, obj)
+        elif isinstance(obj, str):
+            obj = unicode(obj, "utf-8")
+        return obj
+
+    def encode_and_submit_results(obj):
+        demisto.results(str_to_unicode(obj))
+
+    def get_cs_error(stderr):
+        return {
+            "Type": entryTypes["error"],
+            "ContentsFormat": formats["text"],
+            "Contents" : stderr
+        } if stderr else None
+
+    def get_cs_status(search_name, status):
+        return {
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['text'],
+            'Contents': 'Search {} status: {}'.format(search_name, status),
+            'EntryContext': {
+                'EWS.ComplianceSearch(val.Name === obj.Name)': {'Name': search_name, 'Status': status}
+            }
+        }
+
+    def create_ps_file(ps_name, ps_content):
```

Comment:
Check if Powershell supports reading PS script from string (not from file).
For parallel execution.

Reply:
Or add guid to files.

Reply:
added guid to files